src/test/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SandboxInterceptorTest.java

```diff
@@ -865,8 +866,18 @@ public void blockedASTTransformsGrab() throws Exception {
     }
 
     private static Object evaluate(Whitelist whitelist, String script) {
-        GroovyShell shell = new GroovyShell(GroovySandbox.createSecureCompilerConfiguration());
-        Object actual = GroovySandbox.run(shell, script, whitelist);
+        CompilerConfiguration cc = GroovySandbox.createSecureCompilerConfiguration();
+        GroovyShell shell = new GroovyShell(cc);
+        try {
+            Field loaderF = GroovyShell.class.getDeclaredField("loader");
+            loaderF.setAccessible(true);
+            ClassLoader loader = GroovyShell.class.getClassLoader();
+            ClassLoader memoryProtectedLoader = new SecureGroovyScript.CleanGroovyClassLoader(GroovySandbox.createSecureClassLoader(loader), cc);
+            loaderF.set(shell, memoryProtectedLoader);
```

Comment:
It turns out that CleanGroovyClassLoader _must_ be the loader used by the shell (and in turn also the script) for fields to be whitelisted correctly in initializers. I think the reason is [what is mentioned in this line](https://github.com/jenkinsci/script-security-plugin/blob/294a80c190adefeace621ff73c74c6d956b403e0/src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/groovy/SecureGroovyScript.java#L417) in CleanGroovyClassLoader's Javadoc. If you do not use this loader, then the new tests in `SandboxInterceptor` fail. I think that means we'll need some kind of similar change [here](https://github.com/jenkinsci/workflow-cps-plugin/blob/d81da7167828987d3d8d41d52617acfbee86705b/src/main/java/org/jenkinsci/plugins/workflow/cps/CpsGroovyShellFactory.java#L114) in workflow-cps to modify the class loader used in Pipeline? Either way it will definitely need to be tested.

Reply:
Ok, workflow-cps already has a copy of `CleanGroovyClassLoader` it uses for `CpsGroovyShell`, so the fix on that side is just to make sure that it adds a `ClassLoaderWhitelist ` for the shell's class loader when parsing the script. See https://github.com/jenkinsci/workflow-cps-plugin/pull/297.